polynote-kernel/src/main/scala/polynote/kernel/util/Nand.scala

```diff
@@ -0,0 +1,62 @@
+package polynote.kernel.util
+
+import scala.util.{Left => LeftEither, Right => RightEither}
+
+/**
+  * Represents a data type that is either an `A` or a `B` or NEITHER. I'm told it's kind of the "dual" of cats.data.Ior.
+  *
+  * 'tis right-biased.
+  *
+  * An instance of `A [[Nand]] B` is one of:
+  *  - `[[Nand.Left Left]][A]`
+  *  - `[[Nand.Right Right]][B]`
+  *  - `[[Nand.Neither Neither]]`
+  *
+  * I basically copied the easiest 5% of the cats.data.Ior implementation and changed the Both to Neither. Hope it works.
+  */
+sealed abstract class Nand[+A, +B] extends Product with Serializable {
```

Comment:
I'm not convinced of the name of this 😛I mean, it's true that it's not left and right. But `Ior` was named because of `Xor`, which was isomorphic to `Either` and was dropped when the latter was made right-biased. Not because of a relation to Boolean logic.

It's basically standing in for `EitherT[Option, A, B]` to save an extra `fold`? Not that there's anything wrong with that, but I'd just call it `OptionEither` or something like that. And consider `EitherT` instead?

Reply:
Heh, but `Nand`'s logic matches perfectly 😛 and I mean if they did have `Xor`... 😉 But yeah, `OptionEither` is probably a better name. Much easier to understand. 

I thought about `EitherT` but: 
1. I wanted to practice defining a data type 😉 (not the best reason do to something I know)
2. I didn't find an easy way to check the `None` case given an `EitherT` (I guess I'd need to `fold` and then `getOrElse` / use the resulting option's `fold` like you say)? Or how to create an empty `EitherT` (probably there's some way to do it that I wasn't able to figure out).

Reply:
Renamed!